src/ReactUpdate.res

```diff
@@ -1,22 +1,27 @@
 open Belt
 
-type rec update<'action, 'state> =
+type actionDispatcher<'action> = 'action => unit
+
+type rec update<'state, 'action> =
   | NoUpdate
   | Update('state)
-  | UpdateWithSideEffects('state, self<'action, 'state> => option<unit => unit>)
-  | SideEffects(self<'action, 'state> => option<unit => unit>)
-and self<'action, 'state> = {
-  send: 'action => unit,
+  | UpdateWithSideEffects('state, self<'state, 'action> => option<unit => unit>)
+  | SideEffects(self<'state, 'action> => option<unit => unit>)
+and self<'state, 'action> = {
+  send: actionDispatcher<'action>,
+  dispatch: actionDispatcher<'action>,
   state: 'state,
 }
-and fullState<'action, 'state> = {
+and fullState<'state, 'action> = {
```

Comment:
No need to invert the params in `self` and `fullState`. Let's roll that back to avoid issues with existing code.

Reply:
@bloodyowl let me know what you think, I first assumed that since the reducer accepts state first and action second, then everything else should too, but if you think it's better to keep these types as it eas, then I will follow.

Reply:
I'd keep it in their original order for people using it. That'd lead to unexpected bugs.